File: android/src/main/java/expo/modules/motionblurview/models/VelocityField.kt
package expo.modules.motionblurview.models

internal data class VelocityField(val map: AffineMap) {
  fun maxSpeed(rect: Bounds): Double = map.maxMagnitude(rect)

  fun interpolated(target: VelocityField, amount: Double): VelocityField =
    VelocityField(map.interpolated(target.map, amount))

  companion object {
    val ZERO = VelocityField(AffineMap.ZERO)
  }
}

File: ios/Models/VelocityField.swift
//
//  VelocityField.swift
//  Pods
//
//  Created by rit3zh CX on 9/23/26.
//

import CoreGraphics

struct VelocityField: Equatable {
  static let zero = VelocityField(map: .zero)
  var map: AffineMap
  func maxSpeed(in rect: CGRect) -> CGFloat {
    map.maxMagnitude(over: rect)
  }
  func interpolated(to target: VelocityField, amount: CGFloat) -> VelocityField {
    VelocityField(map: map.interpolated(to: target.map, amount: amount))
  }
}
